pkg/version/version.go

```diff
@@ -23,36 +23,45 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
+const (
+	pipelinesControllerSelector string = "app.kubernetes.io/part-of=tekton-pipelines,app.kubernetes.io/component=controller,app.kubernetes.io/name=controller"
+)
+
 // GetPipelineVersion Get pipeline version, functions imported from Dashboard
 func GetPipelineVersion(c *cli.Clients) (string, error) {
-	version := ""
-	namespaces := []string{"tekton-pipelines", "openshift-pipelines"}
+	deploymentsList, err := getDeployments(c)
 
-	listOptions := metav1.ListOptions{
-		LabelSelector: "app.kubernetes.io/component=controller,app.kubernetes.io/name=tekton-pipelines",
+	if err != nil {
+		return "", err
 	}
 
-	var deployments []v1.Deployment
-	for _, namespace := range namespaces {
-		d, _ := c.Kube.AppsV1().Deployments(namespace).List(listOptions)
-		deployments = append(deployments, d.Items...)
-	}
+	version := findVersion(deploymentsList.Items)
 
-	if len(deployments) > 0 {
-		version = findVersion(deployments)
+	if version == "" {
+		return "", fmt.Errorf("Error getting the tekton pipelines deployment version. Version is unknown")
 	}
 
-	if version == "" {
-		d, _ := c.Kube.AppsV1().Deployments("").List(listOptions)
-		deployments = append(deployments, d.Items...)
-		version = findVersion(deployments)
+	return version, nil
+}
+
+// Get deployments for either Tekton Triggers, Tekton Dashboard or Tekton Pipelines
+func getDeployments(c *cli.Clients) (*v1.DeploymentList, error) {
+	deployments, err := c.Kube.AppsV1().Deployments("").List(metav1.ListOptions{LabelSelector: pipelinesControllerSelector})
+	if err != nil {
+		return nil, err
 	}
 
-	if version == "" {
-		return "", fmt.Errorf("Error getting the tekton pipelines deployment version. Version is unknown")
+	// NOTE: If the new labels selector returned nothing, try with old labels selector
+	// The old labels selectors are deprecated and should be removed at some point
+	if deployments == nil || len(deployments.Items) == 0 {
+		appSelector := "app.kubernetes.io/component=controller,app.kubernetes.io/name=tekton-pipelines"
```

Comment:
maybe you can put that as const like we do for `pipelinesControllerSelector` as `oldpipelinesControllerSelector`

Reply:
Also, this will fail for previous version of tekton wouldn't it ? (`app.kubernetes.io` labels are in 0.12 not before, right ?)

Reply:
@vdemeester if the new selector returns nothing, we try with the old selector as a fallback

Reply:
@chmouel done